### ISSUE TITLE
proxy: kind create cluster fails

### ISSUE THREAD

**Issue by ddziuda** [kubernetes-sigs/kind#1579]
**What happened**:
`kind create cluster` fails with:
```
 ✗ Starting control-plane 🕹️
ERROR: failed to create cluster: failed to init node with kubeadm: command "docker exec --privileged kind-control-plane kubeadm init --ignore-preflight-errors=all --config=/kind/kubeadm.conf --skip-token-print --v=6" failed with error: exit status 1
```

**What you expected to happen**:
kind cluster be created

**How to reproduce it (as minimally and precisely as possible)**:
Run `kind create cluster`.

**Anything else we need to know?**:
[kind-export-logs.tar.gz](https://github.com/kubernetes-sigs/kind/files/4611586/kind-export-logs.tar.gz) attached. I'm behind a proxy but I hope I set it correctly for kind.

**Environment:**
- kind version (use `kind version`): kind v0.8.1 go1.14.2 linux/amd64
- Kubernetes version: (use `kubectl version`): 1.18
- Docker version: (use `docker info`): 19.03.2
- OS (e.g. from `/etc/os-release`): CentOS Linux 7 (Core)

I can see in `containerd.log`:
```
May 11 14:56:18 kind-control-plane containerd[259]: time="2020-05-11T14:56:18.637679949Z" level=error msg="Failed to loa
d cni configuration" error="cni config load failed: no network config found in /etc/cni/net.d: cni plugin not initialize
d: failed to load cni config"
```
I viewed the list of open and closed issues but I didn't find any which helped me.

**Comment by BenTheElder** [comment]
you'll have to work around with `NO_PROXY=kind-control-plane` (and possibly the docker subnet), we just patched this in HEAD.

if you checkout this repo and `make build` you'll have a kind binary in `./bin/kind` (you just need make / posix sh / docker) that you can install. if you have GOPATH/bin you can `make install`.

**Comment by ddziuda** [comment]
I think I tried with `NO_PROXY=kind-control-plane` but let me try again.

**Comment by ddziuda** [comment]
Ok, this works for me:
in `/etc/systemd/system/docker.service.d/service-overrides.conf`:
```
[Service]
...
Environment="NO_PROXY=localhost,10.96.0.1,192.168.91.132,172.18.0.1,172.18.0.
2,kind-control-plane,172.17.0.1,172.17.0.2"
```
and `echo $NO_PROXY`:
```
127.0.0.1,localhost,dockerhost,10.96.0.1,192.168.91.132,172.18.0.1,172.18.0.2,172.17.0.1,172.17.0.2,kind-control-plane
```
Maybe I used CIDR notation before and that's why it didn't work.

Now I got successful:
```
[iid@iid kind]$ kind create cluster
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.18.2) 🖼 
 ✓ Preparing nodes 📦  
 ✓ Writing configuration 📜 
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
Set kubectl context to "kind-kind"
You can now use your cluster with:

kubectl cluster-info --context kind-kind

Not sure what to do next? 😅  Check out https://kind.sigs.k8s.io/docs/user/quick-start/
```
but after `kubectl cluster-info --context kind-kind` I'm getting:
```
To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
Error from server (InternalError): an error on the server ("") has prevented the request from succeeding
```
Let me check other open/closed issues for that.

**Comment by ddziuda** [comment]
So it looks similar to:
> Failing to properly start cluster
but on the other hand:
```
[iid@iid kind]$ grep eviction logs/kind-control-plane/kubelet.log 
May 12 10:35:13 kind-control-plane kubelet[1516]: E0512 10:35:13.098679    1516 eviction_manager.go:255] eviction manager: failed to get summary stats: failed to get node info: node "kind-control-plane" not found
May 12 10:35:23 kind-control-plane kubelet[1516]: E0512 10:35:23.099506    1516 eviction_manager.go:255] eviction manager: failed to get summary stats: failed to get node info: node "kind-control-plane" not found
```
and I can also see in the `kubelet.log`:
```
May 12 10:35:22 kind-control-plane kubelet[1516]: E0512 10:35:22.368393    1516 event.go:260] Server rejected event '&v1
.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"kind-control-plane.160e417a88d22623"
, GenerateName:"", Namespace:"default", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time
{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeco
nds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReferenc
e(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, InvolvedObject:v1.ObjectR
eference{Kind:"Node", Namespace:"", Name:"kind-control-plane", UID:"kind-control-plane", APIVersion:"", ResourceVersion:
"", FieldPath:""}, Reason:"NodeHasSufficientPID", Message:"Node kind-control-plane status is now: NodeHasSufficientPID",
 Source:v1.EventSource{Component:"kubelet", Host:"kind-control-plane"}, FirstTimestamp:v1.Time{Time:time.Time{wall:0xbfa
6bcd842b89c23, ext:6456675240, loc:(*time.Location)(0x4158aa0)}}, LastTimestamp:v1.Time{Time:time.Time{wall:0xbfa6bcd846
1a607b, ext:6513414139, loc:(*time.Location)(0x4158aa0)}}, Count:3, Type:"Normal", EventTime:v1.MicroTime{Time:time.Time
{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, Series:(*v1.EventSeries)(nil), Action:"", Related:(*v1.ObjectReference)(n
il), ReportingController:"", ReportingInstance:""}': 'namespaces "default" not found' (will not retry!)
```
I'm attaching [kind-export-logs-after-successful-kind-create-cluster.tar.gz](https://github.com/kubernetes-sigs/kind/files/4615124/kind-export-logs-after-successful-kind-create-cluster.tar.gz) - maybe you can help in this case too.

**Comment by BenTheElder** [comment]
can you take a look at using the latest code in the master branch as mentioned above?

**Comment by BenTheElder** [comment]
I'm pretty sure this was already fixed in HEAD, please follow up if that's not the case.

**Comment by ddziuda** [comment]
I can confirm HEAD really works:
```
[iid@iid kind-sources]$ kind version
kind v0.9.0-alpha go1.14.3 linux/amd64
...
[iid@iid kind-sources]$ kind create cluster
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.18.2) 🖼 
 ✓ Preparing nodes 📦  
 ✓ Writing configuration 📜 
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
Set kubectl context to "kind-kind"
You can now use your cluster with:

kubectl cluster-info --context kind-kind

Thanks for using kind! 😊
[iid@iid kind-sources]$ kind get clusters
kind
[iid@iid kind-sources]$ docker ps
CONTAINER ID        IMAGE                  COMMAND                  CREATED              STATUS              PORTS                       NAMES
4177804dc986        kindest/node:v1.18.2   "/usr/local/bin/entr…"   About a minute ago   Up 46 seconds       127.0.0.1:46597->6443/tcp   kind-control-plane
[iid@iid kind-sources]$ more ~/.kube/config 
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: ...
    server: https://127.0.0.1:46597
  name: kind-kind
contexts:
- context:
    cluster: kind-kind
    user: kind-kind
  name: kind-kind
current-context: kind-kind
kind: Config
users:
- name: kind-kind
  user:
    client-certificate-data: ...
    client-key-data: ...
[iid@iid kind-sources]$ kind get clusters
kind
[iid@iid kind-sources]$ kubectl config current-context
kind-kind
[iid@iid kind-sources]$ kubectl cluster-info
Kubernetes master is running at https://127.0.0.1:46597
KubeDNS is running at https://127.0.0.1:46597/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
```
Thanks.

**Comment by BenTheElder** [comment]
🎉
Thank you!
We have a release approximately scheduled monthly now, will include this in
one later this month.
